### ISSUE TITLE
Load parquet files to a duckdb file

### ISSUE THREAD

**Issue by bendnorman** [catalyst-cooperative/pudl#3739]

**Comment by bendnorman** [comment]
Duckdb table names have a character limit of 63. We have four tables that exceed 63 characters:

`core_eiaaeo__yearly_projected_fuel_cost_in_electric_sector_by_type`: 66
`core_eiaaeo__yearly_projected_generation_in_electric_sector_by_technology`: 73
`core_eiaaeo__yearly_projected_generation_in_end_use_sectors_by_fuel_type`: 72
`out_eia923__yearly_generation_fuel_by_generator_energy_source_owner`: 67

We should rename these resources, enforce the resource name length constraint earlier in the code and update our documentation.

**Comment by cmgosnell** [comment]
hhhhmmm long aeo names. @jdangerx made an [aeo schema pr](https://github.com/catalyst-cooperative/pudl/pull/3588) before migrating a lot of the AEO tables. I think the trouble here is that there are sooooo many AEO tables and so many of them contain the same pieces of information just broken down by different attributes.

**Comment by zaneselvans** [comment]
On the topic of making a DuckDB schema with our metadata classes, I'd been thinking we either want to have `to_sqlite()` and `to_duckdb()` methods in place of the generic `to_sql()` we have now (which only creates SQLite schemas) or maybe add `dialect="duckdb"` and `dialect="sqlite"` arguments to `to_sql()` that do the right thing, and have it default to `dialect="sqlite"` since that's the legacy behavior.

**Comment by bendnorman** [comment]
Agreed! That's what I'm working on right now. I've added a `dialect="duckdb"` option to `to_sql()`. For now, I'm just going to add some if statements to handle the different dialects, but there is probably a cleaner way to store the metadata to SQL logic of multiple dialects. I might make a `MetadataSQLConverter` class or type.

**Comment by zaneselvans** [comment]
It might also be possible to use SQLAlchemy for this -- if the checks, constraints, etc can be stated using their generic API, and then output to the appropriate dialect. IIRC there was at least one SQLite specific thing that we had to code manually though.

**Comment by jdangerx** [comment]
In our inframundo meeting we decided that we can skip the "hard" ones for now and get back to them before we actually release to the public:

* tables that have long names
* CEMS, which is too big to load all at once

**Comment by zaneselvans** [comment]
Something weird is going on with how big the DuckDB file is. Parquet with snappy compression is expected to be about the same size as the compressed DuckDB file. In Parquet, PUDL only takes up like 1-2GB (minus CEMS), and the DuckDB file is like 13GB, which just seems totally wacked.

**Comment by bendnorman** [comment]
I think Duckdb uses a different compression algorithm so duckdb files aren't expected to be as small as parquet files: https://github.com/duckdb/duckdb/discussions/8162#discussioncomment-6371454

**Comment by zaneselvans** [comment]
A factor of 10 feels suspicious though. I searched around for comparisons of the DuckDB and Parquet compression ratios and even [a couple of years ago](https://duckdb.org/2022/10/28/lightweight-compression.html) it looked like DuckDB should be less than 2x as big as Parquet.

**Comment by bendnorman** [comment]
Hmm I thought it could be that we're not specifying varchar lengths but the docs say [that shouldn't matter](https://duckdb.org/docs/sql/data_types/text.html#specifying-a-length-limit).

It looks like many blocks in our `out_eia__monthly_generators` table are uncompressed:

```
D select compression, count(*) as count from pragma_storage_info('out_eia__monthly_generators') group by compression order by count desc;
┌──────────────┬───────┐
│ compression  │ count │
│   varchar    │ int64 │
├──────────────┼───────┤
│ Uncompressed │  4205 │
│ RLE          │  2714 │
│ Constant     │  1719 │
│ Dictionary   │  1281 │
│ FSST         │   722 │
│ ALPRD        │   182 │
│ BitPacking   │    51 │
│ ALP          │    35 │
└──────────────┴───────┘
```

Maybe our indexes are taking up a lot of space?

**Comment by cmgosnell** [comment]
closed bc we stopped this and migrated to marimo

**Comment by zaneselvans** [comment]
I still think we should probably do this at some point (along with writing all the FERC DBs to DuckDB instead of SQLite so they can be queried remotely) though obviously not in the context of Superset.
